### PR TITLE
Homework 1

### PR DESCRIPTION
**Pull request by tribaeka**

### CODE REVIEW

File: homework-1/src/Point.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export class Point {
+    public x: number;
+    public y: number;
+
+    constructor();
+    constructor(x?: number, y?: number) {
```
Comment (AnastasiaGritsuk) [inline homework-1/src/Point.ts]: You can use newer notation like constructor(public x = 0, public y = 0) {}. In this case you don't need explicitly declare x, y.

File: homework-1/src/Point.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.y = y || 0;
+    }
+
+    toString(): string {
+        return `(${this.x}, ${this.y})`;
+    }
+
+    distance(): number;
+    distance(x?: Point): number;
+    distance(x?: any, y?: number): number {
```
Comment (AnastasiaGritsuk) [inline homework-1/src/Point.ts]: Please avoid 'any' type. In this case x : Point | number

File: homework-1/src/Shape.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+export const DEFAULT_COLOR = 'green';
+
+export abstract class Shape {
+    protected color: string;
+    protected filled: boolean;
+    protected points: Point[];
+
+    constructor(points: Point[]);
+    constructor(points: Point[], color?: string, filled?: boolean) {
```
Comment (AnastasiaGritsuk) [inline homework-1/src/Shape.ts]: According to the Hometask "color" and "filled" should be obligatory for this overloading

File: homework-1/src/Point.ts (hunk 1 above)
Comment (AnastasiaGritsuk) [inline homework-1/src/Point.ts]: According to the Hometask "x" and "y" should be obligatory for this overloading

File: homework-1/src/Point.ts (hunk 2 above)
Comment (tribaeka) [reply homework-1/src/Point.ts]: Fixed

File: homework-1/src/Shape.ts (hunk 3 above)
Comment (tribaeka) [reply homework-1/src/Shape.ts]: Fixed, thanks

File: homework-1/src/Point.ts (hunk 1 above)
Comment (tribaeka) [reply homework-1/src/Point.ts]: Fixed

Comment (tribaeka) [reply homework-1/src/Point.ts]: Fixed. Thanks a lot, didn't know about this useful feature.

**Pull request merged**